### ISSUE TITLE
Add markers from an event doesn't affect the dom

### ISSUE THREAD

**Issue by Fraktalite** [bluehalo/ngx-leaflet#123]
Hi,

I'm having an issue with the marker layer and LeafletEvents. It seems that the binding related to leafletLayers is not updated if the array property is modified from a leaflet event. If this array is updated from an external event (button click) the binding is correctly updated including the operations made previously with the leaflet events. I tried to force the array update by using the slice method you suggest just after the push but it had no effects.

To reproduce it, just take the marker demo app, and modify the code to support adding marker when you click on the map:
`onMapReady(map: Map) {
map.on('click', (e : LeafletMouseEvent) => {
      this.addMarker(e.latlng); 
    });
	}`

Modify the addMarker method as follow:
```
addMarker(latlng: LatLng = latLng(46.879966 + 0.1 * (Math.random() - 0.5), -121.726909 + 0.1 * (Math.random() - 0.5))) {
	const newMarker = marker(
		[ latlng.lat, latlng.lng ],
		{
			icon: icon({
				iconSize: [ 16, 16 ],
				iconAnchor: [ 8, 8 ],
				iconUrl: '2273e3d8ad9264b7daa5bdbf8e6b47f8.png',
				shadowUrl: '44a526eed258222515aa21eaffd14a96.png'
			})
		}
	);
    this.markers.push(newMarker);
    this.markers = this.markers.slice();
}
```

Of course, don't forget to update html with the event:

```
<div leaflet style="height: 300px;"
	[leafletOptions]="options"
	[leafletLayers]="markers"
	(leafletMapReady)="onMapReady($event)">
</div>
```

What you will see is when you click on the addMarker button, the marker is correctly added to the map at a random position, just like before. If you click on the map, nothing happen, if you look at the DOM the marker has not been added. Then if you click again on the button you will see two new marker: the new random one and a marker located where you click on the map.

### In summary:

1. click on the addMarker button: _1 marker is on the map_
2. click on the map: _still 1 marker on the map_
3. click on the addMarker button: _3 markers are on the map_

Hope I'm clear, otherwise let me know.

**Comment by Tommatheussen** [comment]
Actually, I wanted to just report this as well, but as described on the README: https://github.com/Asymmetrik/ngx-leaflet#a-note-about-change-detection
Leaflet events do not trigger Angular change detection, adding a manual trigger for angular to detect changes (like the example) worked perfectly for me.

**Comment by reblace** [comment]
@Fraktalite let me know if the above works for you. If it doesn't, I can try to reproduce your specific issue. You can also see https://github.com/Asymmetrik/ngx-leaflet/issues/118 for some of the discussion around this issue.

**Comment by Fraktalite** [comment]
Thanks to you. this fix my issues. I was waiting to make some tests with data from a REST api but this definetely fix the problem.

what I don't know is how much it's expensive to call the change detection manually.

Thanks again

**Comment by reblace** [comment]
In response to the "expensive to call change detection manually"  question - it isn't really expensive if you're only doing it in response to some rare (i.e., only when a user clicks) event. It's only really expensive if you're doing it constantly (i.e., every time the mouse moves or every 10 ms or something).

Angular already does change detection in response to a lot of user interactions, so you will unlikely notice a difference in your specific use case of only triggering it on map click events.

Glad your issue's resolved!
